### PR TITLE
Deprecate depends_on in favor of dependson

### PR DESCRIPTION
**Pull request by olblak**
Signed-off-by: Olblak <me@olblak.com>

Align depends_on syntax with the current manifest convention.
This also remove the confusion introduce by jsonschema

## Test

To test this pull request, you can run the following commands:

```shell
cp <to_package_directory>
go test
```

### CODE REVIEW

File: pkg/core/pipeline/resource/main.go [hunk 1]
```diff
@@ -42,6 +42,8 @@ type ResourceConfig struct {
 	SCMID string `yaml:",omitempty"` // SCMID references a uniq scm configuration
 	// !deprecated, please use scmid
 	DeprecatedSCMID string `yaml:"scmID,omitempty" jsonschema:"-"` // SCMID references a uniq scm configuration
+	// !deprecated, please use scmid
```
Comment (lemeurherve) [inline pkg/core/pipeline/resource/main.go:45]:
```suggestion
	// !deprecated, please use dependson
```

File: pkg/core/pipeline/resource/main.go [hunk 2]
```diff
@@ -27,7 +27,7 @@ import (
 
 type ResourceConfig struct {
 	// depends_on specifies which resources must be executed before the current one
```
Comment (lemeurherve) [inline pkg/core/pipeline/resource/main.go:29]:
```suggestion
	// dependson specifies which resources must be executed before the current one
```

File: pkg/core/pipeline/resource/main.go [hunk 3]
```diff
@@ -42,6 +42,8 @@ type ResourceConfig struct {
 	SCMID string `yaml:",omitempty"` // SCMID references a uniq scm configuration
 	// !deprecated, please use scmid
 	DeprecatedSCMID string `yaml:"scmID,omitempty" jsonschema:"-"` // SCMID references a uniq scm configuration
+	// !deprecated, please use scmid
+	DeprecatedDependsOn []string `yaml:"depends_on,omitempty" jsonschema:"-"` // SCMID references a uniq scm configuration
```
Comment (lemeurherve) [inline pkg/core/pipeline/resource/main.go:46]:
```suggestion
	DeprecatedDependsOn []string `yaml:"depends_on,omitempty" jsonschema:"-"` // depends_on specifies which resources must be executed before the current one
```

File: pkg/core/pipeline/condition/main.go [hunk 4]
```diff
@@ -134,6 +134,19 @@ func (c *Config) Validate() error {
 		}
 	}
 
+	// Handle depends_on deprecation
+	if len(c.DeprecatedDependsOn) > 0 {
+		switch len(c.DependsOn) == 0 {
+		case true:
+			logrus.Warningf("%q is deprecated in favor of %q.", "depends_on", "dependson")
```
Comment (lemeurherve) [inline pkg/core/pipeline/condition/main.go:141]: Why the interpreted string here? I would have used a simple string, it won't change.
(Same below and in the other files)

**Review by lemeurherve — COMMENTED** [review]
Can the depends_on deprecation be handled in a single place instead of being duplicated 3 times?

File: pkg/core/pipeline/condition/main.go (hunk 4 above)
Comment (olblak) [reply pkg/core/pipeline/condition/main.go:141]: It's more a matter of taste, I prefer the readability this way than something like 

`logrus.Warningln("\"depends_on\" is deprecated in favor of \"dependson\".")`

**Comment by olblak** [comment]
> Can the depends_on deprecation be handled in a single place instead of being duplicated 3 times?

Well we could refactor the code to do so but I don't think it's worth doing now.

**Pull request merged**